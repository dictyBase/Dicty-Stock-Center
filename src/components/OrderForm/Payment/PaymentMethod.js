// @flow
import React, { useState } from "react"
import { Grid } from "@material-ui/core"
import RadioGroup from "@material-ui/core/RadioGroup"
import Radio from "@material-ui/core/Radio"
import FormControlLabel from "@material-ui/core/FormControlLabel"
import TextField from "../TextField"
import RequiredTextLabel from "../RequiredTextLabel"
import useStyles from "../formStyles"

type Props = {
  /** Function for handling radio button selection */
  handleChange: Function,
  /** Function to manually set Formik field values */
  setFieldValue: Function,
}

/**
 * PaymentMethod contains radio buttons and a text field for listing payment method information.
 */

const PaymentMethod = ({ handleChange, setFieldValue }: Props) => {
  const [purchaseOrderNum, setPurchaseOrderNum] = useState(false)
  const classes = useStyles()

  return (
    <Grid
      container
      justify="center"
      alignItems="center"
      className={classes.innerForm}>
      <RequiredTextLabel title="Payment Account" />
      <Grid item xs={12} md={8}>
        <RadioGroup
          aria-label="Payment Method"
          name="paymentMethod"
          onChange={handleChange}
          row>
          <FormControlLabel
            value="credit"
            control={<Radio />}
            label="Credit Card"
            onChange={() => {
              setPurchaseOrderNum(false)
              setFieldValue("purchaseOrderNum", "N/A - credit card")
            }}
          />
          <FormControlLabel
            value="wire"
            control={<Radio />}
            label="Wire transfer"
            onChange={() => {
              setPurchaseOrderNum(false)
              setFieldValue("purchaseOrderNum", "N/A - wire transfer")
            }}
          />
          <FormControlLabel
            value="purchaseOrder"
            control={<Radio />}
            label="Purchase Order (PO)"
            onChange={() => setPurchaseOrderNum(true)}
          />
        </RadioGroup>
        {purchaseOrderNum && (
          <TextField
            type="text"
            name="purchaseOrderNum"
            placeholder="PO Number"
          />
        )}
      </Grid>
    </Grid>
  )
}

export default PaymentMethod
