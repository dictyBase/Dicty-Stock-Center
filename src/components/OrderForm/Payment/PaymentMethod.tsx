import React, { useState } from "react"
import { Grid } from "@material-ui/core"
import TextField from "../TextField"
import RequiredTextLabel from "../RequiredTextLabel"
import useStyles from "../formStyles"
import PaymentMethodRadioGroup from "./PaymentMethodRadioGroup"

type Props = {
  /** Function for handling radio button selection */
  handleChange: () => void
  /** Function to manually set Formik field values */
  setFieldValue: Function
}

/**
 * PaymentMethod contains radio buttons and a text field for listing payment method information.
 */

const PaymentMethod = ({ handleChange, setFieldValue }: Props) => {
  const [purchaseOrderNum, setPurchaseOrderNum] = useState(false)
  const classes = useStyles()

  return (
    <Grid
      container
      justify="center"
      alignItems="center"
      className={classes.innerForm}>
      <RequiredTextLabel title="Payment Account" />
      <Grid item xs={12} md={8}>
        <PaymentMethodRadioGroup
          setPurchaseOrderNum={setPurchaseOrderNum}
          handleChange={handleChange}
          setFieldValue={setFieldValue}
        />
        {purchaseOrderNum && (
          <TextField
            type="text"
            name="purchaseOrderNum"
            placeholder="PO Number"
          />
        )}
      </Grid>
    </Grid>
  )
}

export default PaymentMethod
