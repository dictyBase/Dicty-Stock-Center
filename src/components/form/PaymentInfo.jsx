import React, { Component, PropTypes } from 'react'
import { reduxForm } from 'redux-form'
import Consumer from './Consumer'
import User from './User'
import 'styles/core.scss'

export const fields = [ 'firstName', 'lastName', 'email', 'org', 'group',
    'address', 'address2', 'city', 'state', 'zip', 'country', 'phone',
    'shipAccount', 'shipAccountNum', 'comments' ]

class PaymentInfo extends Component {
    displayName = 'payment information'

    static propTypes = {
        order: PropTypes.object
    }

    render() {
        const { consumer } = this.props.order
        const { editShipping } = this.props.orderActions
        const {
            fields: { firstName, lastName, email, org, group, address, address2, city,
                state, zip, country, phone }
        } = this.props
        return (
            <div>
                <div className="row">
                    <div className="col-xs-12">
                        <h2 className="page-header">Please enter payment information</h2>
                    </div>
                </div>
                <div className="row">
                    <div className="col-xs-12">
                        <Consumer consumer={ consumer } edit={ editShipping } />
                    </div>
                </div>
                <hr />
                <div className="row">
                    <form onSubmit={ '' } className="form-horizontal">
                        <div className="col-md-6">
                            <User title = { 'Payer Address' }
                                firstName = { firstName }
                                lastName = { lastName }
                                email = { email }
                                org ={ org }
                                group = { group }
                                address = { address }
                                address2 = { address2 }
                                city = { city }
                                state = { state }
                                zip = { zip }
                                country = { country }
                                phone = { phone }
                            />
                        </div>
                        <div className="col-md-6">
                            Payment Method
                        </div>
                    </form>
                </div>
            </div>
        )
    }
}

export default reduxForm({
    form: 'payment',
    fields
})(PaymentInfo)

