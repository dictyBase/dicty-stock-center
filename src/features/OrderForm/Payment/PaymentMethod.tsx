import React, { useState } from "react"
import { makeStyles, Theme } from "@material-ui/core/styles"
import { grey } from "@material-ui/core/colors"
import Box from "@material-ui/core/Box"
import Typography from "@material-ui/core/Typography"
import TextField from "../TextField"
import RequiredTextLabel from "../RequiredTextLabel"
import PaymentMethodRadioGroup from "./PaymentMethodRadioGroup"

const useStyles = makeStyles((theme: Theme) => ({
  waiver: {
    backgroundColor: grey[200],
    border: "1px solid #e6f2ff",
    borderRadius: theme.spacing(1),
    margin: theme.spacing(2),
    padding: theme.spacing(3),
  },
}))

/**
 * PaymentMethod contains radio buttons and a text field for listing payment method information.
 */

const PaymentMethod = () => {
  const [purchaseOrderNum, setPurchaseOrderNum] = useState(false)
  const [waiverRequested, setWaiverRequested] = useState(false)
  const classes = useStyles()

  return (
    <Box mt={1} mb={2} p={2}>
      <RequiredTextLabel title="Payment Account" variant="h3" />
      <Box mt={1} />
      <PaymentMethodRadioGroup
        setPurchaseOrderNum={setPurchaseOrderNum}
        setWaiverRequested={setWaiverRequested}
      />
      {purchaseOrderNum && (
        <TextField name="purchaseOrderNum" placeholder="PO Number" />
      )}
      {waiverRequested && (
        <Typography component="p" className={classes.waiver}>
          Please send an email to{" "}
          <a href="mailto:dictystocks@northwestern.edu" target="_top">
            dictystocks@northwestern.edu
          </a>{" "}
          describing why you need a waiver.
        </Typography>
      )}
    </Box>
  )
}

export default PaymentMethod
