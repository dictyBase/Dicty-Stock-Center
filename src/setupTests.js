import { configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import "jest-enzyme"

global.console = {
  log: jest.fn(),
  error: jest.fn(),
  warn: jest.fn(),
}

configure({ adapter: new Adapter() })
