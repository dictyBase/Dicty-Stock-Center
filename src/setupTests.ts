import { configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import "jest-enzyme"
import { sprintf } from "sprintf-js"

let globalAny = global as any

globalAny.console = {
  log: (msg: string, args: any) => {
    if (typeof msg !== "string") {
      return
    }
    const str = sprintf(msg, args)
    process.stderr.write(str + "\n")
  },
  error: jest.fn(),
  warn: jest.fn(),
}

configure({ adapter: new Adapter() })
