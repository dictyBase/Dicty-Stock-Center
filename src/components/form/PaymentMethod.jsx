import React, { Component, PropTypes } from 'react'
import Panel from 'dicty-react-components/src/Panel'
import PanelHeader from 'dicty-react-components/src/PanelHeader'
import PanelTitle from 'dicty-react-components/src/PanelTitle'
import PanelBody from 'dicty-react-components/src/PanelBody'
import PaymentInfo from './PaymentInfo'
import 'styles/core.scss'

export default class PaymentMethod extends Component {
    displayName = 'payment method';

    static propTypes = {
        payMethod: PropTypes.object.isRequired,
        poNum: PropTypes.object.isRequired,
        title: PropTypes.string.isRequired
    }

    render() {
        const { payMethod, poNum, title } = this.props
        return (
            <Panel>
                <PanelHeader>
                   <PanelTitle>{ title }</PanelTitle>
                </PanelHeader>
                <PanelBody>
                    <PaymentInfo payMethod={ payMethod } poNum={ poNum }/>
                </PanelBody>
            </Panel>
        )
    }
}
