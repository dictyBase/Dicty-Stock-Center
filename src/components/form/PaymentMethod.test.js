import React from "react"
import "jest-styled-components"
import "../../setupTests"
import PaymentMethod from "./PaymentMethod"
import PaymentInfo from "./PaymentInfo"
import { shallow } from "enzyme"

describe("form/PaymentMethod", () => {
  const payMethod = { value: "credit", touched: true }
  const poNum = { value: "", touched: false }
  const title = "Method"

  const wrapper = shallow(
    <PaymentMethod payMethod={payMethod} poNum={poNum} title={title} />,
  )

  it("should render <PaymentInfo>", () => {
    expect(
      wrapper.contains(<PaymentInfo payMethod={payMethod} poNum={poNum} />),
    ).toEqual(true)
  })
})
