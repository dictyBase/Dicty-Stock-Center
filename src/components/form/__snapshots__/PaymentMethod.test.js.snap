// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of PaymentMethod 1`] = `
<div
  className="gsdTnU"
  style={
    Object {
      "border": "1px solid #D2D7D3",
      "height": "100%",
      "marginBottom": 0,
    }
  }
>
  <div
    className="bBgvkp"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#337ab7",
        "borderBottom": "0px none #ddd",
        "borderColor": "#4B77BE",
        "padding": "20px",
      }
    }
  >
    <h3
      className="jXCrp"
      style={
        Object {
          "color": "#ffffff",
          "fontSize": "20px",
          "fontWeight": "200",
        }
      }
    >
      <div
        className="kyzudI"
        onClick={[Function]}
      >
        Method
      </div>
    </h3>
  </div>
  <div
    className="panel-body cgBkWQ"
    open={true}
    style={Object {}}
  >
    <div
      className="DRXhs"
    />
    <div
      className="sc-ipXKqB eOLIpB"
    >
      <div
        className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
        is={null}
      >
        <div
          className="sc-ifAKCX cFlEyZ sc-bdVaJa cyncUw"
          is={null}
        >
          <label
            className="sc-VJcYb mTwLq"
          >
            <span
              className="sc-giadOv eAuPGE"
              title="required field"
            >
              * 
            </span>
            Payment Method:
          </label>
        </div>
        <div
          className="sc-ifAKCX cFlEyZ sc-bdVaJa XupaC"
          is={null}
        >
          <label
            className="sc-qrIAp hsouou"
          >
            <input
              checked={false}
              touched={true}
              type="radio"
              value="Credit card"
            />
              Credit Card
          </label>
          <label
            className="sc-qrIAp hsouou"
          >
            <input
              checked={false}
              touched={true}
              type="radio"
              value="Wire transfer"
            />
              Wire Transfer
          </label>
          <label
            className="sc-qrIAp hsouou"
          >
            <input
              checked={false}
              touched={true}
              type="radio"
              value="PO"
            />
              Purchase Order (PO)
          </label>
        </div>
      </div>
    </div>
    <div
      className="DRXhs"
    />
  </div>
</div>
`;
